source/advanced-topics.adoc

```diff
@@ -1143,3 +1143,77 @@ The `%ifos` conditional is used to control processing based on the operating sys
   %endif
 
 All the contents of the SPEC file between `%ifos` and `%endif` are processed only if the build was done on a Linux system.
+
+[id="Packaging-of-Python-RPMs"]
+=== Packaging of Python RPMs
+
+[id="Benefits-of-packaging-Python-packages-as-RPMs"]
+==== Benefits of packaging Python packages as RPMs
+
+Benefits of packaging Python packages as RPMs are the same as for any other RPM package. More can be found in the chapter xref:Why-Package-Software-with-RPM[].
+
+[id="Converting-PyPI-package-to-an-RPM"]
+==== Converting PyPI package to an RPM
+
+Pyp2rpm is a tool which automatically converts PyPI packages to an RPM spec file. Simple tutorial can be found on this page link:https://github.com/fedora-python/pyp2rpm[pyp2rpm].
+
+[id="Typical-Python-RPM-package-spec-file-description"]
+==== Typical Python RPM package spec file description
+
+There is a chapter xref:what-is-a-spec-file[], which describes spec file in general. Python RPM spec file specifications to spec files shown already in this guide might be shown in this example:
+
+[source,specfile]
+----
+Name:           python-setuptools
+Version:        0.9.8
+Release:        3%{?dist}
+Summary:        Easily build and distribute Python packages
+License:        Python or ZPLv2.0
+URL:            https://pypi.org/project/setuptools/
+Source0:        https://files.pythonhosted.org/packages/source/s/setuptools/setuptools-%{version}.zip
+
+BuildArch:      noarch
+
+BuildRequires:  python2-devel
+
+BuildRequires:  python-backports-ssl_match_hostname
+Requires:         python-backports-ssl_match_hostname
+
+%description
+Setuptools is a collection of enhancements to the Python distutils that allow
+you to more easily build and distribute Python packages, especially ones that
+have dependencies on other packages.
+
+This package contains the runtime components of setuptools, necessary to
+execute the software that requires pkg_resources.py.
+
+%prep
+%setup -q -n setuptools-%{version}
+
+%build
+CFLAGS="$RPM_OPT_FLAGS" %{__python} setup.py build
+
+%install
+%{__python} setup.py install --skip-build --root %{buildroot}
+
+%check
+LANG=en_US.utf8 %{__python} setup.py test
+
+%files
+%doc *.txt docs
+%{python_sitelib}/*
+%{_bindir}/easy_install
+
+%changelog
+...
+----
+
+[id="Common-macros-for-Python-RPM-packages"]
+==== Common macros for Python RPM packages
+
+All Python specific RPM macros can be found on this page link:https://fedoraproject.org/wiki/Packaging:Python#Macros[Python packaging macros].
```

Comment:
Again, the spec file above is RHEL 6/7, but the link is for Fedora. That is very different currently and rather confusing.

Reply:
Are there at least some macros the same?
Anyway, we might pick at least the macros from the example, and either use the description from Fedora, or write a new one. I don't think we need as comprehensive guide for RHEL, as we have for Fedora, a simple variant should work.
Another thought regarding this is that Fedora macros will be relevant for the next major version of RHEL, so it's also question what RHEL version this guide is focused on..

Reply:
Not really. They all got python version number in Fedora.

Reply:
What RHEL version we target has never been answered to me.

Reply:
It is written for RHEL7 (RHEL version) and for Fedora (community version) under conditionals. Other versions of RHEL might be written in different paragraphs stating for which version they are written in their headings. The above link to macros was only WIP. I would like to see several most used and useful macros for RHEL7 and for Fedora we do not need to document them all.